promql/engine.go

```diff
@@ -394,24 +395,10 @@ func (ng *Engine) newQuery(q storage.Queryable, expr Expr, start, end time.Time,
 	return qry
 }
 
-// testStmt is an internal helper statement that allows execution
-// of an arbitrary function during handling. It is used to test the Engine.
-type testStmt func(context.Context) error
```

Comment:
I don't think this belongs to the parser

Reply:
Not really.

However, `testStmt` is of type `Statement`, which can only be implemented in the parser package because of https://github.com/slrtbtfs/prometheus/blob/split_parser/promql/parser/ast.go#L54 .


The alternative to what I'm doing would be making `Statement.stmt()` public, so `testStmt` could live in the engine, but that kind of defeats the point of  `Statement.stmt()`.

Since having one more `Statement` type in the parser doesn't really hurt I decided to go that way.